model/modeldecoder/v2/decoder.go

```diff
@@ -1434,3 +1443,19 @@ func mapSpanLinks(from []spanLink, out *[]model.SpanLink) {
 		}
 	}
 }
+
+func inferFromResource(res nullable.String) *model.ServiceTarget {
+	outTarget := &model.ServiceTarget{}
+	if res.IsSet() {
+		submatch := patternWithSingleSlash.FindStringSubmatch(res.Val)
+		if submatch != nil {
+			outTarget.Type = submatch[1]
+			outTarget.Name = submatch[2]
+		} else if patternWithOnlyAlphabets.MatchString(res.Val) {
+			outTarget.Type = res.Val
+		} else {
+			outTarget.Name = res.Val
+		}
+	}
+	return outTarget
+}
```

Comment:
What do you think about simplifying this to a single regex expression, accepting a string, and not returning a pointer to that struct:

```suggestion
func targetFromDestinationResource(resource string) (target model.ServiceTarget) {
	submatch := patternServiceTargetRe.FindStringSubmatch(resource)
	switch len(submatch) {
	case 1:
		target.Type = resource
	case 2:
		target.Type = submatch[1]
		target.Name = submatch[2]
	default:
		target.Name = resource
	}
	return target
}
```

Reply:
> accepting a string, and not returning a pointer to that struct
Sounds cleaner, will do the changes

> What do you think about simplifying this to a single regex expression
Let's continue the discussion in the lines involving regex comment below since it will depend on the conclusion of using a single regex expression